tracing-subscriber/src/registry/fmt_layer.rs

```diff
@@ -0,0 +1,197 @@
+use crate::{
+    fmt::{format, FormatEvent, FormatFields, MakeWriter},
+    layer::{Context, Layer},
+    registry::{Extensions, LookupMetadata, LookupSpan, Registry, SpanData, SpanRef},
+};
+use ansi_term::{Color, Style};
+use humantime;
+use std::{
+    any::type_name,
+    fmt::{self, Write as _},
+    io::{self, Write},
+    marker::PhantomData,
+    time::SystemTime,
+};
+use tracing_core::{
+    field::{Field, Visit},
+    span::{Attributes, Id},
+    Event, Level, Subscriber,
+};
+
+pub struct FmtLayer<
+    S = Registry,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    is_interested: Box<dyn Fn(&Event<'_>) -> bool + Send + Sync + 'static>,
```

Comment:
hmmm, I'd prefer not to _force_ boxing here (esp. since we _don't_ box anything else). Note that there is [an impl of `Fn` for `Box<F>` where `F: Fn`](https://doc.rust-lang.org/std/ops/trait.Fn.html#implementors), so this could be a generic parameter with a _default_ type of `Box<dyn Fn(&Event<'_>) -> bool + Send + Sync + 'static>`.

Reply:
also, in an ideal world, i'd really prefer it if this layer didn't do filtering of its own, and was composed with a separate filter --- see https://github.com/tokio-rs/tracing/issues/302. this is fine for now, though, and when we implement #302, we can remove it.

Reply:
+1 to relying on #302.

I applied your suggestion, and I'd love to delegate that as a per-layer filter mechanism. The number of generic bounds is getting a _bit_ uncomfortable and is defeating the purpose of layers. Each generic bound should probably be factored out into its own layer (except for the `FormatEvent` + `FormatFields` bounds, which should be together).

Reply:
well, tried that, but the type inference wasn't great. reverted & adding a todo.